akka-remote/src/main/scala/akka/remote/artery/ImmutableLongMap.scala

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import scala.annotation.tailrec
+import akka.util.OptionVal
+import scala.reflect.ClassTag
+import java.util.Arrays
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ImmutableLongMap {
+  def empty[A >: Null](implicit t: ClassTag[A]): ImmutableLongMap[A] =
+    new ImmutableLongMap(Array.emptyLongArray, Array.empty)
```

Comment:
It's not like we call empty in a hot path with the current usage, so it probably does not matter.